### PR TITLE
feat(ingress-setup): add section for ingress controllers

### PR DESCRIPTION
**Pull request by nloding**
## Description

Add short section giving a bit more detail around ingress controllers. We received a couple community questions recently, and this little bit of information may have helped them self-service and understand their issue better.

## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [ ] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [x] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [x] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by nloding** [comment]
@jessesimpson36 tagging you in for a review!

**Comment by jessesimpson36** [comment]
@nloding Thanks so much for the PR. I already see it as a huge help.

 I do have a couple notes about this:

The [nginx ingress controller](https://github.com/kubernetes/ingress-nginx/blob/main/README.md#readme) that you linked is actually Not the same nginx ingress controller  in the helm install command.  I certainly would not expect you to have noticed this since one project calls itself ingress-nginx and the other project calls itself nginx-ingress.  This link should be changed to https://docs.nginx.com/nginx-ingress-controller/installation/installation-with-helm/  .

> If you are using a cloud provider such as AWS or GCP, we recommend you use follow their ingress setup guides.

I believe cloud providers provide their own ingress controller by default. I don't know for sure, and I like that this statement is in here.  I just wonder if we should add a  "if necessary" at the end, because some users may not have to perform any sort of setup.

And the last thing, is perhaps we should go through these options we are feeding into the nginx ingress controller just to make sure they're necessary.  Some of those options might just be default values.  I can do this part.

**Comment by akeller** [comment]
Please add @christinaausley as a reviewer of this PR when you are ready. It looks like there is still work in progress.

Also, if this is available but undocumented, it will likely need to be added to other versions (8.2, at least). If you need help with that, please let us know!

**Comment by christinaausley** [comment]
Following up here @nloding -- is this still in progress? If so, please:

- Backport to applicable versions.
- Loop me in for review.

If this is on `hold`, please add the appropriate label with status updates so we can stay on top of open PRs 😄 .

**Comment by jessesimpson36** [comment]
Hey all, I added a commit with some changes I wanted made, and I think this change is ready to backport to some previous versions. 

Before I do, I'd like to know if you all had a backporting script or methodology that would reduce the chance of me making a mistake as I manually edit these files?

**Comment by jessesimpson36** [comment]
The methodology I used just now was 
```
git format-patch HEAD^
patch 0001-feat-ingress-setup-add-section-for-ingress-controlle.patch versioned_docs/version-8.2/self-managed/platform-deployment/helm-kubernetes/guides/ingress-setup.md
```

This could be formatted into a shell script, if you did not already have a tool for doing this.

**Comment by jessesimpson36** [comment]
@christinaausley , please review the changes.

**Review by jessesimpson36 — DISMISSED** [review]

**Comment by akeller** [comment]
> Hey all, I added a commit with some changes I wanted made, and I think this change is ready to backport to some previous versions.
> 
> Before I do, I'd like to know if you all had a backporting script or methodology that would reduce the chance of me making a mistake as I manually edit these files?

@pepopowitz see this comment and the one below from @jessesimpson36. We have #1779.

**Comment by nloding** [comment]
Gah, sorry all - I got distracted on some other work and forgot to circle back. I just pushed a small change to fix a typo in the new content, but I'm good with the rest!

File: docs/self-managed/platform-deployment/helm-kubernetes/guides/ingress-setup.md [hunk 1]
```diff
@@ -185,6 +185,29 @@ Once deployed, you can access the Camunda Platform 8 components on:
 - **Keycloak authentication:** `https://keycloak.camunda.example.com`
 - **Zeebe Gateway:** `grpc://zeebe.camunda.example.com`
 
+## Ingress Controllers
+
+Ingress resources require the cluster to have an [Ingress Controller](https://kubernetes.io/docs/concepts/services-networking/ingress-controllers/) running. There are many options for configuring your Ingress Controller. If you are using a cloud provider such as AWS or GCP, we recommend you follow their Ingress setup guides if an Ingress Controller is not already pre-installed.
```
Comment (christinaausley) [inline docs/self-managed/platform-deployment/helm-kubernetes/guides/ingress-setup.md:190]: May want to provide links to such guides for AWS and GCP.

**Review by christinaausley — APPROVED** [review]
One non-blocking comment 👍

File: docs/self-managed/platform-deployment/helm-kubernetes/guides/ingress-setup.md (hunk 1 above)
Comment (nloding) [reply docs/self-managed/platform-deployment/helm-kubernetes/guides/ingress-setup.md:190]: I wasn't able to find official guides. There is discussion of ingress controllers in the middle of their load balancing documentation, and some outside blog posts, but nothing I felt was concrete enough to link to from our official documentation. Curious if @jessesimpson36 has any resources here? Could be I just got lost in their documentation and missed it!

Comment (christinaausley) [reply docs/self-managed/platform-deployment/helm-kubernetes/guides/ingress-setup.md:190]: @jessesimpson36 Let me know if you have preferences here -- can always adjust in a subsequent PR.

**Pull request merged**